CRM/Relationshipblock/Page/Inline/RelationshipBlock.php

```diff
@@ -10,6 +10,12 @@ public function run() {
       return;
     }
     self::addKeyRelationshipsBlock($this, $contactId);
+
+    // check logged in user permission
+    if (!isset($page->_permission)) {
+      CRM_Contact_Page_View::checkUserPermission($this, $contactId);
+    }
```

Comment:
What is the purpose of wrapping this line in a conditional? In other similar inline block pages it's not. E.g. 

https://github.com/civicrm/civicrm-core/blob/bc77d7c0798081f999e6eca92d75fa874fd04e74/CRM/Contact/Page/Inline/Email.php#L58-L57

Reply:
Thanks for reviewing @colemanw.

This is to prevent it being called during initial load of the contact summary screen; since it's already going to be called for every other similar inline block page, *and* the page itself.
Seeing as how `checkUserPermission` doesn't cache its results (or more accurately, doesn't use the results it's cached), and potentially reaches out to the database one or more times when it's called, it seemed like a fair optimisation.

Reply:
@agileware-fj - can you add a code comment to that effect & point to a gitlab (erm which you will log) - that way when the right fix is done the line can be removed again